Build sporadically fails due to `make` remains

https://travis-ci.com/volvo-cars/signalbroker-server/jobs/280434302

```
CMake Error: The current CMakeCache.txt directory /signalbroker/apps/app_unixds/priv/CMakeCache.txt is different than the directory /home/travis/build/volvo-cars/signalbroker-server/apps/app_unixds/priv where CMakeCache.txt was created. This may result in binaries being created in the wrong place. If you are not sure, reedit the CMakeCache.txt
CMake Error: The source "/signalbroker/apps/app_unixds/c_lib/CMakeLists.txt" does not match the source "/home/travis/build/volvo-cars/signalbroker-server/apps/app_unixds/c_lib/CMakeLists.txt" used to generate cache.  Re-run cmake with a different source directory.
c_lib/elixir.mk:6: recipe for target 'priv/testunit' failed
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -19,6 +19,7 @@ before_install:
   - mix local.rebar --force
   - mix deps.get
   - mix test --only success
+  - mix clean
   - docker build -t signalbroker-server:latest -f ./docker/Dockerfile .
   # - echo "testing inside docker container..."
   # - docker run -d -p 127.0.0.1:80:4567 carlad/sinatra /bin/sh -c "cd /root/sinatra; bundle exec foreman start;"

